fuel-txpool/src/service.rs

```diff
@@ -1,143 +1,137 @@
-use std::cmp::Reverse;
+use crate::{interface::Interface, types::*, Config};
+use fuel_core_interfaces::block_importer::ImportBlockBroadcast;
+use fuel_core_interfaces::model::TxInfo;
+use fuel_core_interfaces::txpool::{TxPoolDb, TxPoolMpsc, TxStatusBroadcast};
+use parking_lot::Mutex as BlockingMutex;
 use std::sync::Arc;
-
-use crate::Error;
-use crate::{subscribers::MultiSubscriber, types::*, Config, TxPool as TxPoolImpl};
-use async_trait::async_trait;
-use fuel_core_interfaces::model::{ArcTx, TxInfo};
-use fuel_core_interfaces::txpool::{Subscriber, TxPool, TxPoolDb};
-use std::collections::HashMap;
-use tokio::sync::RwLock;
-
-pub struct TxPoolService {
-    txpool: RwLock<TxPoolImpl>,
-    db: Box<dyn TxPoolDb>,
-    subs: MultiSubscriber,
+use tokio::sync::{broadcast, mpsc, oneshot, Mutex};
+use tokio::task::JoinHandle;
+use tracing::warn;
+
+pub struct Service {
+    interface: Arc<Interface>,
+    sender: mpsc::Sender<TxPoolMpsc>,
+    broadcast: broadcast::Sender<TxStatusBroadcast>,
+    join: BlockingMutex<Option<JoinHandle<mpsc::Receiver<TxPoolMpsc>>>>,
+    receiver: Arc<Mutex<Option<mpsc::Receiver<TxPoolMpsc>>>>,
 }
 
-impl TxPoolService {
-    pub fn new(db: Box<dyn TxPoolDb>, config: Config) -> Self {
-        Self {
-            txpool: RwLock::new(TxPoolImpl::new(config)),
-            db,
-            subs: MultiSubscriber::default(),
-        }
+impl Service {
+    pub fn new(db: Box<dyn TxPoolDb>, config: Config) -> Result<Self, anyhow::Error> {
+        let (sender, receiver) = mpsc::channel(100);
+        let (broadcast, _receiver) = broadcast::channel(100);
```

Comment:
if we want the channels to be recyclable (due to service restart etc), maybe we should consolidate the initialization of all of them into the main fuel-core service?

Reply:
It is the same problem, if I want to receive from `Receiver` i need a `mut` version of it, I could wrap it inside Mutex but this is similar to what I am doing now

Reply:
You are now talking about full restart of fuel-client? The better way is to kill the app and start it again as it will be the most usual and most tested way how you start the client.